code/lib/core-server/src/utils/StoryIndexGenerator.test.ts

```diff
@@ -0,0 +1,1468 @@
+/* eslint-disable @typescript-eslint/no-shadow */
+/// <reference types="@types/jest" />;
+
+/**
+ * @jest-environment node
+ */
+
+import path from 'path';
+import fs from 'fs-extra';
+import { normalizeStoriesEntry } from '@storybook/core-common';
+import type { Indexer, NormalizedStoriesSpecifier, StoryIndexEntry } from '@storybook/types';
+import { loadCsf, getStorySortParameter } from '@storybook/csf-tools';
+import { toId } from '@storybook/csf';
+import { logger, once } from '@storybook/node-logger';
+
+import type { StoryIndexGeneratorOptions } from './StoryIndexGenerator';
+import { StoryIndexGenerator } from './StoryIndexGenerator';
+
+jest.mock('@storybook/csf-tools');
+jest.mock('@storybook/csf', () => {
+  const csf = jest.requireActual('@storybook/csf');
+  return {
+    ...csf,
+    toId: jest.fn(csf.toId),
+  };
+});
+
+jest.mock('@storybook/node-logger');
+
+const toIdMock = toId as jest.Mock<ReturnType<typeof toId>>;
+const loadCsfMock = loadCsf as jest.Mock<ReturnType<typeof loadCsf>>;
+const getStorySortParameterMock = getStorySortParameter as jest.Mock<
+  ReturnType<typeof getStorySortParameter>
+>;
+
+const storiesMdxIndexer: Indexer = {
+  test: /\.stories\.mdx$/,
+  index: async (fileName, opts) => {
+    let code = (await fs.readFile(fileName, 'utf-8')).toString();
+    const { compile } = await import('@storybook/mdx2-csf');
+    code = await compile(code, {});
+    const csf = loadCsf(code, { ...opts, fileName }).parse();
+
+    // eslint-disable-next-line no-underscore-dangle
+    return Object.entries(csf._stories).map(([key, story]) => ({
+      key,
+      id: story.id,
+      name: story.name,
+      title: csf.meta.title,
+      importPath: fileName,
+      type: 'story',
+      tags: story.tags ?? csf.meta.tags,
+    }));
+  },
+};
+
+const csfIndexer: Indexer = {
+  test: /\.stories\.(m?js|ts)x?$/,
+  index: async (fileName, options) => {
+    const code = (await fs.readFile(fileName, 'utf-8')).toString();
+    const csf = loadCsf(code, { ...options, fileName }).parse();
+
+    // eslint-disable-next-line no-underscore-dangle
+    return Object.entries(csf._stories).map(([key, story]) => ({
+      key,
+      id: story.id,
+      name: story.name,
+      title: csf.meta.title,
+      importPath: fileName,
+      type: 'story',
+      tags: story.tags ?? csf.meta.tags,
+    }));
+  },
+};
+
+const options: StoryIndexGeneratorOptions = {
+  configDir: path.join(__dirname, '__mockdata__'),
+  workingDir: path.join(__dirname, '__mockdata__'),
+  storyIndexers: [],
+  indexers: [csfIndexer, storiesMdxIndexer],
+  storiesV2Compatibility: false,
+  storyStoreV7: true,
+  docs: { defaultName: 'docs', autodocs: false },
+};
+
```

Comment:
none of the tests here are new, they've just been copied back from the deprecated test suite. The only thing new here is the indexers above